core/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java

```diff
@@ -79,6 +78,7 @@
         private int bulkActions = 1000;
         private ByteSizeValue bulkSize = new ByteSizeValue(5, ByteSizeUnit.MB);
         private TimeValue flushInterval = null;
+        private BackoffPolicy backoffPolicy = BackoffPolicy.noBackoff();
```

Comment:
To me, "default listener implementation" just meant that we need an implementation that is capable of doing that without having the user write one. The behavior of not backing off was chosen because it was the behavior up to now and the user can easily change it via a new builder method (user docs follow soon). 

But I can change the default behavior to exponential backoff quite simply. It's a one line change (apart from updating Javadocs appropriately).